### PR TITLE
cleanup extension settings formatting

### PR DESCRIPTION
**Pull request by mgreenegit**
Request to merge changes to simplify passing properties to the DSC extension

### CODE REVIEW

**Comment by mgreenegit** [comment]
#sign-off @bmoore-msft

File: 101-automation-configuration/nested/provisionServer.json [hunk 1]
```diff
@@ -49,7 +50,7 @@
     {
       "name": "[parameters('automationAccountName')]",
       "type": "Microsoft.Automation/automationAccounts",
-      "apiVersion": "2018-01-15",
+      "apiVersion": "[variables('automationApiVersion')]",
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionServer.json]: revert - leave this as a literal value so schema validation can be done

File: 101-automation-configuration/nested/provisionServer.json [hunk 2]
```diff
@@ -123,47 +124,21 @@
...
-              {
-                "Name": "ActionAfterReboot",
-                "Value": "ContinueConfiguration",
-                "TypeName": "System.String"
+        },
+        "protectedSettings": {
+          "configurationArguments": {
+              "registrationKey": {
+                "userName": "notused",
+                "Password": "[listKeys(resourceId('Microsoft.Automation/automationAccounts/', parameters('automationAccountName')), variables('automationApiVersion')).Keys[0].value]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionServer.json]: same as above - use a literal apiVersion

File: 101-automation-configuration/nested/provisionServer.json (hunk 1 above)
Comment (mgreenegit) [reply 101-automation-configuration/nested/provisionServer.json]: ack

**Pull request merged**